src/Analyser/TypeSpecifier.php

```diff
@@ -156,14 +156,17 @@ public function specifyTypesInCondition(
 			}
 
 			$classType = $scope->getType($expr->class);
-			$type = TypeTraverser::map($classType, static function (Type $type, callable $traverse): Type {
+			$uncertainty = false;
+			$type = TypeTraverser::map($classType, static function (Type $type, callable $traverse) use (&$uncertainty): Type {
```

Comment:
I tried to copy the logic from MutatingScope::inferType with an uncertainty boolean:
https://github.com/phpstan/phpstan-src/blob/adcabb3078603f46c811c7dc460174c9df551310/src/Analyser/MutatingScope.php#L960-L989